src/scales/scale.time.js

```diff
@@ -261,7 +261,7 @@ function determineUnit(minUnit, min, max, capacity) {
 		interval = INTERVALS[UNITS[i]];
 		factor = interval.steps ? interval.steps[interval.steps.length - 1] : MAX_INTEGER;
 
-		if (Math.ceil((max - min) / (factor * interval.size)) <= capacity) {
+		if (Math.ceil((max - min) / (factor * interval.size)) <= capacity && INTERVALS[UNITS[i]].common) {
```

Comment:
Better to reuse the previous object access and check a boolean before calling `Math.ceil`:

`if (interval.common && Math.ceil((max - m...`

Reply:
good call. i've updated it